content/blog/modern-text-features/index.Rmd

```diff
@@ -0,0 +1,381 @@
+It is both possible to force even-width numbers, as well as correct display of fractional numbers (using `font_feature(numbers = c("tabular", "fractions")`) using OpenType as long as the font supports it, so it is definitely something to look into when you want to add that final polish to your visualization.
+
+## Color fonts
+
+A recent (in font technology terms) development is the availability of color fonts, i.e. fonts where the glyphs have designated colors.
+This development is largely driven by the ubiquity of emojis in modern text, and while it may seem that emojis have been around forever, it is recent enough that the world has yet to converge to a single standard for color fonts.
+The system emoji font on macOS, Windows, and Linux all uses different font technologies for storing the color glyphs, ranging from storing a single bitmap, to storing each glyph as an SVG.
+This, unsurprisingly, complicates things.
+To add insult to injury, emojis often gets rendered slightly larger than the surrounding text and with a slightly lowered baseline in a very OS-specific way (this does not apply to all color fonts; only emojis).
+
+Why am I telling you this?
+Well, honestly it is mostly to make you appreciate the labor that went into the fact that color fonts (and by extension, emojis) now just works:
+
+```{r, eval=FALSE}
+emojis <- "👩🏾‍💻🔥📊"
+
+p <- ggplot() + 
+  geom_label(
+    aes(x = 0, y = 0, label = emojis), 
+    family = "Apple Color Emoji"
+  )
+
+preview_devices(p, "emoji")
+```
+
+```{r, echo=FALSE, results='asis'}
+get_results("emoji")
+```
+
+As one can see the failures range from not being able to render anything, to rendering in monochrome.
+Further, it appears as if the devices have trouble figuring out the dimensions of the glyphs.
+One additional wrinkle is that while Cairo on macOS is capable of rendering in monochrome, it fails to get the correct emoji.
+This is because emojis relies heavily on ligatures, and the "dark-skinned woman at a computer" emoji is actually a ligature of the "woman", "dark skin" and "computer" emojis.
+
+## Font fallback
+
+In all of the above examples we have been very mindful in setting the font-face to a font that contains all the glyphs we need.
+This is not always practical, especially when you want to mix emoji and regular text.
+It is also an absolute requirement when mixing Latin and CJK (Chinese, Japanese, and Korean) text as it is infeasible to include all CJK glyphs in a single font.
+However, we are used to things just working at the system level.
+No matter the font it seems that a glyph is always displayed in e.g. browsers and text editors.
+This is because the OS is employing font fallback, which is the act of figuring out an alternative font to use when a glyph is not present in the chosen font.
```

Comment:
```suggestion
This is because the OS is employing **font fallback**, which is the act of figuring out an alternative font to use when a glyph is not present in the chosen font.
```